students/carlos_novoa/lesson09/mailroom.py

```diff
@@ -0,0 +1,337 @@
+            print(rf.format(*args))
+
+    def donor_summary(self, name):
+        summary = ('\n\n===========================\n'
+                   'Donor: {}\n'
+                   'Total Donations: {}.\n'
+                   'Donation Count: {}\n'
+                   'Donation Average: {}\n'
+                   '===========================\n\n')
+        total = self.donations_total(name)
+        count = self.donations_count(name)
+        average = self.donations_average(name)
+        print(summary.format(name, total, count, average))
+
+    def donations_total(self, name=None):
+        return Helpers.format_currency_str(sum(self.donors[name]))
+
+    def donations_count(self, name=None):
+        return len(self.donors[name])
+
+    def donations_average(self, name=None):
+        total = sum(self.donors[name])
+        count = len(self.donors[name])
+        return Helpers.format_currency_str(total / count)
+
+
+class Donor(Donors):
+    donors = None
+
+    def __init__(self, donors=None):
+        super().__init__(donors)
+
+    def donors_update(self, name, donation):
+        """ Update base donor dictionary """
+        if self.donor_key_found(name) is False:
+            # new donor
+            self.donors[name] = [float(donation)]
+        else:
+            # existing donor
+            self.donors[name].append(float(donation))
+
+    @staticmethod
+    def donor_email(name, donation):
+        donation = Helpers.format_currency_str(donation)
+        m = ('\n\nDear {},\n\n'
+             '        Thank you for your very kind donation of {}.\n\n'
+             '        It will be put to very good use.\n\n'
+             '               Sincerely,\n'
+             '                  -The Team\n\n')
+        print(m.format(name, donation))
+
+
+class MailroomCli:
+
+    @staticmethod
+    def prompt_name():
+        while True:
+            name = input('Type donor name: ')
+            try:
+                if name is None:
+                    raise ValueError()
+            except ValueError:
+                print('Name string required.')
+                continue
+            return name
+
+    @staticmethod
+    def prompt_donation():
+        while True:
+            donation = input('Donation amount: ')
+            try:
+                donation = float(donation)
+            except ValueError:
+                print('Input must be a int/float, try again.')
+                continue
+            return donation
+
+    def process_new_donor(self):
+        D = Donors()
+        d = Donor()
+
+        while True:
+            name = self.prompt_name()
+            if D.donor_key_found(name) is False:
+                break
+            else:
+                print('Donor already exists, use append.')
+
+        donation = self.prompt_donation()
+        self.thank_donor(name, donation)
+        d.donors_update(name, donation)
+
+    def process_donation(self):
+        D = Donors()
+        d = Donor()
```

Comment:
This is pretty confusing to use 'D' and 'd' as variable names and also a good indication that there isn't enough separating the roles of these classes in your code. See my comment above on reworking these classes for better encapsulation and separation of concerns